examples/distributed/pyg/distributed_cpu.py

```diff
@@ -0,0 +1,497 @@
+            loss = train_fn(batch)
+            result = (f'[Node {dist_context.rank}] Train: '
+                      f'epoch={epoch}, it={i}, loss={loss:.4}, '
+                      f'time={(time.time() - batch_time):.4}')
+            batch_time = time.time()
+            if logfile:
+                log = open(logfile, 'a+')
+                log.write(f'{result}\n')
+                log.close()
+            if not progress_bar:
+                print(result)
+
+    torch.distributed.barrier()
+    return loss
+
+
+def run_proc(
+    local_proc_rank: int,
+    num_nodes: int,
+    node_rank: int,
+    dataset: str,
+    dataset_root_dir: str,
+    master_addr: str,
+    ddp_port: int,
+    train_loader_port: int,
+    test_loader_port: int,
+    num_epochs: int,
+    batch_size: int,
+    num_neighbors: str,
+    async_sampling: bool,
+    concurrency: int,
+    num_workers: int,
+    num_loader_threads: int,
+    progress_bar: bool,
+    logfile: str,
+):
+    if dataset == 'ogbn-mag':
+        is_hetero = True
+    elif dataset == 'ogbn-products':
+        is_hetero = False
+    else:
+        raise NotImplementedError(f'This example supports only OGB datasets: '
+                                  f'(ogbn-products, ogbn-mag), got {dataset}')
+
+    print('--- Loading data partition files ...')
+    root_dir = osp.join(osp.dirname(osp.realpath(__file__)), dataset_root_dir)
+    node_label_file = osp.join(root_dir, f'{dataset}-label', 'label.pt')
+    train_idx = torch.load(
+        osp.join(
+            root_dir,
+            f'{dataset}-train-partitions',
+            f'partition{node_rank}.pt',
+        ))
+    test_idx = torch.load(
+        osp.join(
+            root_dir,
+            f'{dataset}-test-partitions',
+            f'partition{node_rank}.pt',
+        ))
+
+    if is_hetero:
+        train_idx = ('paper', train_idx)
+        test_idx = ('paper', test_idx)
+
+    # load partition information
+    (
+        meta,
+        num_partitions,
+        partition_idx,
+        node_pb,
+        edge_pb,
+    ) = load_partition_info(osp.join(root_dir, f'{dataset}-partitions'),
+                            node_rank)
+    print(f'meta={meta}, partition_idx={partition_idx}')
+    # load partition into graph
+    graph = LocalGraphStore.from_partition(
+        osp.join(root_dir, f'{dataset}-partitions'), node_rank)
+    # load partition into feature
+    feature = LocalFeatureStore.from_partition(
+        osp.join(root_dir, f'{dataset}-partitions'), node_rank)
+
+    # setup the partition information in LocalGraphStore and LocalFeatureStore
+    graph.num_partitions = feature.num_partitions = num_partitions
+    graph.partition_idx = feature.partition_idx = partition_idx
+    graph.node_pb = feature.node_feat_pb = node_pb
+    graph.edge_pb = feature.edge_feat_pb = edge_pb
+    graph.meta = feature.meta = meta
+    feature.labels = torch.load(node_label_file)
+    partition_data = (feature, graph)
```

Comment:
This part isn't very clear to me and might be also confusing for a user. Can't we specify this via `from_partition` calls?

Reply:
You're right, I've overlooked this step. I moved `load_partition_info` to LGS/LFS `from_partition`